st2api/setup.py

```diff
@@ -14,27 +14,43 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-try:
-    from setuptools import setup, find_packages
-except ImportError:
-    from ez_setup import use_setuptools
-    use_setuptools()
-    from setuptools import setup, find_packages
+import re
+import os.path
+from pip.req import parse_requirements
+from setuptools import setup, find_packages
+
+
+def fetch_requirements():
+    links = []
+    reqs = []
```

Comment:
Some requirements are represented as git+proto//URL@branch#some_egg_dep or
other (of course I hope not to see svn in there :) these will be stored in
links array.

The links array is given to setup as a separate argument (if I'm not wrong
`dependency_list`. When setuptools want to locate `some_egg_dep` it will
look through the links array first and only after fallback to PyPI.

The reqs array contains `someeggg, =ver` requirements and similar.

On Tue, Jun 23, 2015, 08:01 Lakshmi Kannan <notifications@github.com> wrote:

> In st2api/setup.py
> <https://github.com/StackStorm/st2/pull/1647#discussion_r33009564>:
>
> > @@ -14,27 +14,43 @@
> >  # See the License for the specific language governing permissions and
> >  # limitations under the License.
> >
> > -try:
> > -    from setuptools import setup, find_packages
> > -except ImportError:
> > -    from ez_setup import use_setuptools
> > -    use_setuptools()
> > -    from setuptools import setup, find_packages
> > +import re
> > +import os.path
> > +from pip.req import parse_requirements
> > +from setuptools import setup, find_packages
> > +
> > +
> > +def fetch_requirements():
> > +    links = []
> > +    reqs = []
>
> What is this short form for?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StackStorm/st2/pull/1647/files#r33009564>.
>